game/src/sandbox/speed.rs

```diff
@@ -140,12 +145,104 @@ impl SpeedControls {
                 .build_widget(ctx, "reset to midnight"),
         );
 
-        self.panel = Panel::new(Widget::custom_row(row))
-            .aligned(
-                HorizontalAlignment::Center,
-                VerticalAlignment::BottomAboveOSD,
-            )
-            .build(ctx);
+        let mut panel = Panel::new(Widget::col(vec![
+            self.create_time_panel(ctx, app).named("time"),
+            Widget::custom_row(row),
+        ]))
+        .aligned(HorizontalAlignment::Left, VerticalAlignment::Top);
+        if let Some(h) = self.override_height {
+            panel = panel.exact_height(h);
+        }
+        self.panel = panel.build(ctx);
+    }
+
+    fn create_time_panel(&mut self, ctx: &EventCtx, app: &App) -> Widget {
+        let (finished, unfinished) = app.primary.sim.num_trips();
+        let trip_results = Widget::row(vec![
+            {
+                let mut txt = Text::new();
+                let pct = if unfinished == 0 {
+                    100.0
+                } else {
+                    100.0 * (finished as f64) / ((finished + unfinished) as f64)
+                };
+                txt.add(Line(format!(
+                    "Finished trips: {} ({}%)",
+                    prettyprint_usize(finished),
+                    pct as usize
+                )));
+                txt.into_widget(ctx).centered_vert()
+            },
+            if app.primary.dirty_from_edits {
+                ctx.style()
+                    .btn_plain
+                    .icon("system/assets/tools/warning.svg")
+                    .build_widget(ctx, "see why results are tentative")
+                    .centered_vert()
+                    .align_right()
+            } else {
+                Widget::nothing()
+            },
+        ]);
+        // TODO This likely fits better in the top center panel, but no easy way to squeeze it
+        // into the panel for all gameplay modes
+        let record_trips = if let Some(n) = app.primary.sim.num_recorded_trips() {
```

Comment:
This is indeed weird, but it's a pretty obscure dev only feature, so whatever.

Maybe with the upcoming "move everything into the panels" it'd make sense to also move these controls into the panel.

Reply:
Maybe so. I think it could come in handy for any challenge mode, scenario, or freeform... meaning every GameplayMode needs to know about it. Right now there's no centralized handling of `top_right`. Maybe it'll make sense to do that at some point.